packages/core/src/simulate.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  AnyActorContext,
+  AnyActorLogic,
+  AnyStateMachine,
+  EventFromLogic,
+  InternalStateFrom
+} from '.';
+import { createEmptyActor } from './actors';
+
+function createMockActorContext(): AnyActorContext {
+  const emptyActor = createEmptyActor();
+  return {
+    self: emptyActor,
+    logger: console.log,
+    id: 'root_test',
+    sessionId: Math.random().toString(32).slice(2),
+    defer: () => {},
+    system: emptyActor,
+    stopChild: () => {}
+  };
+}
+
+export function simulate<T extends AnyActorLogic>(
+  actorLogic: T,
+  options: { input?: any } = {}
+) {
+  const dummyActorContext = createMockActorContext();
+
+  const sim = {
+    transition: (
+      state: InternalStateFrom<T> = actorLogic.getInitialState(
+        dummyActorContext,
+        options.input
+      ),
+      event: EventFromLogic<T>
+    ): InternalStateFrom<T> => {
+      return actorLogic.transition(state, event, dummyActorContext);
+    },
+    getInitialState: () => {
+      return actorLogic.getInitialState(dummyActorContext, options.input);
+    },
+    microstep: (
```

Comment:
Im not sure if this should be here. It’s specific to machines and it’s not even expressed at type level correctly~ right now. Maybe we can just squint are eyes and accept this duck typing but it would be cool if we wouldnt have to do it

Reply:
Removed